### PR TITLE
here geocoder url config parameter

### PR DESCRIPTION
**Pull request by jcardus**
Looks like it was hardcoded.

The here url has changed.

### CODE REVIEW

**Comment by jcardus** [comment]
checkStyle failing...

**Comment by tananaev** [comment]
What's the new URL? I think we should set at least to be a default.

**Comment by jcardus** [comment]
https://reverse.geocoder.api.here.com/6.2/reversegeocode.json

I put it on the unit test.

My api_key doesn't work with the old url.

By the way I'm having problems trying to fix the unit tests, It has to be run with the Conext initialized because of the "concrete map type error". But we need a settings file for that and I think it's too much for such a small test. Any idea on this?

**Comment by jcardus** [comment]
sorry,
this is the new url:

https://reverse.geocoder.ls.hereapi.com/6.2/reversegeocode.json

File: src/main/java/org/traccar/geocoder/HereGeocoder.java [hunk 1]
```diff
@@ -19,20 +19,24 @@
...
-        url += "&prox=%f,%f,0";
-        url += "&app_id=" + id;
-        url += "&app_code=" + key;
-        if (language != null) {
-            url += "&language=" + language;
-        }
-        return url;
+    private static String formatUrl(String url, String id, String key, String language) {
+      if (url == null) {
+        url = "https://reverse.geocoder.ls.hereapi.com/6.2/reversegeocode.json";
```
Comment (tananaev) [inline src/main/java/org/traccar/geocoder/HereGeocoder.java]: Please fix indentation here.

File: src/main/java/org/traccar/geocoder/HereGeocoder.java [hunk 2]
```diff
@@ -19,20 +19,24 @@
...
+      if (language != null) {
+          url += "&language=" + language;
+      }
+      return url;
     }
 
-    public HereGeocoder(String id, String key, String language, int cacheSize, AddressFormat addressFormat) {
-        super(formatUrl(id, key, language), cacheSize, addressFormat);
+    public HereGeocoder(String url, String id, String key, String language,
+                        int cacheSize, AddressFormat addressFormat) {
```
Comment (tananaev) [inline src/main/java/org/traccar/geocoder/HereGeocoder.java]: And here. We don't align parameters with open bracket. See any other classes for formatting examples.

File: src/test/java/org/traccar/geocoder/GeocoderTest.java [hunk 3]
```diff
@@ -73,7 +73,7 @@ public void testBan() {
     @Ignore
     @Test
     public void testHere() {
-        Geocoder geocoder = new HereGeocoder("", "", null, 0, new AddressFormat());
+        Geocoder geocoder = new HereGeocoder("https://reverse.geocoder.ls.hereapi.com/6.2/reversegeocode.json", "", "", null, 0, new AddressFormat());
```
Comment (tananaev) [inline src/test/java/org/traccar/geocoder/GeocoderTest.java]: URL here can be removed.

File: src/test/java/org/traccar/geocoder/GeocoderTest.java [hunk 4]
```diff
@@ -73,7 +73,7 @@ public void testBan() {
     @Ignore
     @Test
     public void testHere() {
-        Geocoder geocoder = new HereGeocoder("", "", null, 0, new AddressFormat());
+        Geocoder geocoder = new HereGeocoder("", "", "", null, 0, new AddressFormat());
```
Comment (tananaev) [inline src/test/java/org/traccar/geocoder/GeocoderTest.java]: This doesn't look right. Shouldn't the URL be `null`?

**Comment by tananaev** [comment]
Merged, thanks.

**Pull request merged**